ABDataCollectionCore.js

```diff
@@ -563,7 +563,7 @@ module.exports = class ABDataCollectionCore extends ABMLClass {
       // and if this data colleciton is not loading all data from the server on init we need to reload data from server
       // finally if we tell the data collection that we should reload the data because of an update we should force the reload
       // the final check happens when a data collection doesn't have a new set cursor but one of its children has updated causing it to be stale (ex: calculated fields)
-      if (!this.settings.loadAll && shouldReloadData) {
+      if (this.__reloadWheres && !this.settings.loadAll && shouldReloadData) {
```

Comment:
OK, to be honest, if I come back 3 months from now, I'm not gonna follow the logic that is happening here.

Partly the issue is: `__reloadWheres`  is just not documented anywhere.  We just reference it and use it.  What does this data represent?  Who has this set, and who doesn't.  If I remember correctly, it should only be on a DC (`child`)  that is Linked to the cursor of another DC (`parent`).  When the parent cursor changes, then we should have a different condition in `__reloadWheres` for us to go and query the server with.
`Is that correct?`   I'm not even sure.

and it is difficult for me to know what `refreshLinkCursor()` is actually supposed to do.  When are we expecting this to be called?  Who calls this?  What is this function attempting to accomplish? 

It takes stepping through the code to figure these things out.  

it would mean something very different to me if this was expected to be called by a `parent` DC so it can go out and tell all it's `child` DCs to update.  As opposed to be called by a `child` DC when it's parent is changed.

I think the second option is closer to the expectation than the first. (maybe?)

Reply:
Good point the only goodish description of what reloadWheres is doing is in a different repository. So we should give a better description in the core so it is clear when and why it is being set.

Reply:
I have pushed up a change that documents more pieces of this process